src/utils/object.js

```diff
@@ -144,12 +145,50 @@ function splitPath (path) {
   return result;
 }
 
+/**
+ * Normalize a sloppy (or dot-separated) path into a "valid" path.
```

Comment:
Did you try [`path.normalize`](https://nodejs.org/api/path.html#path_path_normalize_path) instead?